### PR TITLE
Issue/2997 post preview enhancements

### LINKED ISSUE #2997: Explain what local changes/drafts are on post preview
The phrases "local changes" and "local draft" may be confusing to some users. Since the default action when tapping a post is to show the post preview, and a bar to post preview which explains what a "local draft" is or what "local changes" are.

### PR DESCRIPTION
**Pull request by nbradbury**
Resolves #2997 - Adds a message bar to post preview which enables publishing a local draft, and publishing or reverting a post with local changes.

local-changes

### CODE REVIEW

**Comment by aerych** [comment]
Nice. :)  
Two tiny nitpicks, take 'em or leave 'em. 
1.  We could probably drop the "yet" from the message.
2.  It might sound less technical to say "discard" vs "revert".

**Comment by nbradbury** [comment]
> Two tiny nitpicks, take 'em or leave 'em.

Good call on dropping "yet." I'm not sure about "Discard," though - what are the chances that'd be confused with "Delete?"

**Comment by aerych** [comment]
> what are the chances that'd be confused with "Delete?"

Hmm. Good question.
For kicks I acked WP core for usage of "discard" in user facing text in wp-admin and didn't find any, however there were a couple of cases where "revert" was used. I guess keep it as is for consistency if nothing else.

**Comment by nbradbury** [comment]
> For kicks I acked WP core for usage of "discard" in user facing text in wp-admin and didn't find any, however there were a couple of cases where "revert" was used. I guess keep it as is for consistency if nothing else.

:+1:

**Comment by daniloercoli** [comment]
I did a "block box" review of the patch and seems fine. 
One thing, the analytics event "editor_closed is fired when I click on "discard" or on "publish". This need to be removed.

Going to start the code review.

**Comment by nbradbury** [comment]
> the analytics event "editor_closed is fired when I click on "discard" or on "publish"

Strange, I'm not seeing this. It looks like the only place that's fired is [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L331) in EditPostActivity. Any chance you clicked EDIT while previewing the post?

**Comment by daniloercoli** [comment]
You're right, sorry.  I had modified the post in the editor. 

I'm now seeing another issue with analytics.
If I modify a post without update the changes to the server, right in the editor by pressing the "update" button, the only event that is tracked by the app is `editor_closed`. When I come back later in the new post preview screen and hit the "Publish" button in the message bar there is no `editor_post_updated` event fired.

If I modify the post in the editor and tap the "update" button both the `editor_post_updated` and `editor_closed` are fired.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostPreviewActivity.java [hunk 1]
```diff
@@ -116,17 +158,142 @@ public boolean onOptionsItemSelected(final MenuItem item) {
...
+
+    private void publishPost() {
+        if (!isFinishing() && NetworkUtils.checkConnection(this)) {
+            PostUploadService.addPostToUpload(mPost);
+            startService(new Intent(this, PostUploadService.class));
+        }
+    }
+
+    @SuppressWarnings("unused")
+    public void onEventMainThread(PostEvents.PostUploadStarted event) {
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostPreviewActivity.java]: Are these 2 methods really necessary?
I see the same calls in the UpdatePostTask.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostPreviewActivity.java]: @daniloercoli Are you referring to the two `onEventMainThread` methods? If so, those are necessary in order to update the UI when the post upload has started and ended. `UpdatePostTask` is entirely separate - it simply gets the latest version of the post.

**Comment by nbradbury** [comment]
> When I come back later in the new post preview screen and hit the "Publish" button in the message bar there is no editor_post_updated event fired.

Added in fda166b.

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**